### ISSUE TITLE
photoswipe.js gets broken after passing through Babel compiler

### ISSUE THREAD

**Issue by PanzerKunst** [dimsemenov/PhotoSwipe#1072]
Line 12 of photoswipe.js (`})(this, function () {`) gets replaced by `})(undefined, function () {`, because of this: https://phabricator.babeljs.io/T6783.

Same thing in photoswipe-ui-default.js. I solved the issue by replacing both instances of `this` by `window`.

**Comment by prabhakaran8737** [comment]
@PanzerKunst Does this solution solves?

**Comment by prabhakaran8737** [comment]
Still I could not make it work with 

**Running the code through Reactjs**

```
var box = new PhotoSwipe(pswpElement, PhotoSwipeUI_Default, items, options);
box.init();  // TypeError: box.init is not a function
```

**Comment by dimsemenov** [comment]
This issue will be fixed in the next update.

@prabhakaran8737, it's unlikely that your problem is related to it. If you're using PhotoSwipw with React, this might help – https://github.com/vn38minhtran/react-photoswipe

**Comment by prabhakaran8737** [comment]
@dimsemenov There is some problem while executing photoswipe through babel

**Comment by DenisMir** [comment]
@dimsemenov Will there be an update regarding this issue in the future?

**Comment by ttomdewit** [comment]
Any news on this, @dimsemenov? I'd love to create a PR but wouldn't know where to start. I've tested the workaround provided by @PanzerKunst and can safely say it works _for our case_.

Any update would be greatly appreciated.

If you'd accept a PR where both `photoswipe.js` and `photoswipe-ui-default.js` are updated to use `window` instead of `this`, I'd gladly provide it. Please note that running `grunt` with the current setup caused a few deprecation warnings.

**Comment by jhabdas** [comment]
Sounds like a Babel issue. Please close @PanzerKunst

**Comment by nirfse** [comment]
Apparently, I also came across this issue. Either I ditch off `babel` from `gulp`, nor I get following errors on a web-page:

```
Uncaught TypeError: Cannot set property 'PhotoSwipe' of undefined
    at photoswipe.min.js:4
    at photoswipe.min.js:4
```

**Comment by jhabdas** [comment]
@nirfse Errors like the one mentioned are indicative something is not working properly. If they are occurring as a result of your asset/build pipeline you may want to start there. When the OP opened this issue they linked to a Babel problem. Unfortunately the Phrabricator link provided has rotted.

**Comment by ttomdewit** [comment]
@nirfse @jhabdas I seem to remember there's a Babel setting in your `.babelrc` file that prevents this from happening. I might be mistaken and confuse this with the new `import` and `require` functions.

**Comment by nirfse** [comment]
@jhabdas @ttomdewit Thanks for your replies, I ended up adding this to `.babelrc`:
```
{
  "ignore": [
    "./node_modules/photoswipe/dist/photoswipe.min.js",
    "./node_modules/photoswipe/dist/photoswipe-ui-default.min.js"
  ]
}
```
For some reason, PhotoSwipe was the only problematic library among other sources in my `gulp`.

**Comment by dimsemenov** [comment]
fixed by v5.0
